engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OpenIdConnectService.scala

```diff
@@ -26,13 +27,15 @@ class OpenIdConnectService[
   protected val useIdToken: Boolean = configuration.jwt.exists(_.userinfoFromIdToken)
 
   override def obtainAuthorizationAndUserInfo(authorizationCode: String): Future[(AuthorizationData, Option[UserData])] = {
-    clientApi.accessTokenRequest(authorizationCode).flatMap { authorization =>
-      val userInfoFromIdToken: Option[UserData] =
-        authorization.idToken.filter(_ => useIdToken).flatMap(introspectToken[UserData])
-      userInfoFromIdToken.map(Future(_))
-        .getOrElse(clientApi.profileRequest(authorization.accessToken))
-        .map(userInfo => (authorization, Some(userInfo)))
-    }
+    clientApi.accessTokenRequest(authorizationCode)
+      .andThen { case Success(authorization) if accessTokenIsJwt => introspectAccessToken(authorization.accessToken) }
+      .flatMap { authorization =>
+        (if (useIdToken) {
+          introspectJwtToken[UserData](authorization.idToken.get)
+        } else {
+          clientApi.profileRequest(authorization.accessToken)
```

Comment:
Can u write one line why we obtain user info and it makes that authorization works correctly?